test/views/UnauthorisedViewSpec.scala

```diff
@@ -16,47 +16,83 @@
-      val link = view.getElementById("get_eori_link")
+  "Unauthorised Page view" when {
+    val view: Html = getUnauthorisedPageView(false)
 
-      link must containMessage("unauthorised.paragraph.1.bullet.1.link")
-      link must haveHref("https://www.gov.uk/eori")
-      link.attr("target") mustBe "_self"
-    }
+    "TdrUnauthorisedMessage is Disabled" should {
+      "display page header" in {
+        view.getElementsByTag("h1").first() must containMessage("unauthorised.heading")
+      }
+
+      "display get EORI link" in {
+        val link = view.getElementById("get_eori_link")
+
+        link must containMessage("unauthorised.paragraph.1.bullet.1.link")
+        link must haveHref("https://www.gov.uk/eori")
+        link.attr("target") mustBe "_self"
+      }
+
+      "display access CDS link" in {
+        val link = view.getElementById("access_cds_link")
 
-    "display access CDS link" in {
-      val link = view.getElementById("access_cds_link")
+        link must containMessage("unauthorised.paragraph.1.bullet.2.link")
+        link must haveHref("https://www.gov.uk/guidance/get-access-to-the-customs-declaration-service")
+        link.attr("target") mustBe "_self"
+      }
 
-      link must containMessage("unauthorised.paragraph.1.bullet.2.link")
-      link must haveHref("https://www.gov.uk/guidance/get-access-to-the-customs-declaration-service")
-      link.attr("target") mustBe "_self"
+      "display check CDS application status link" in {
+        val link = view.getElementById("check_cds_application_status_link")
+
+        link must containMessage("unauthorised.paragraph.2.link")
+        link must haveHref("https://www.tax.service.gov.uk/customs/register-for-cds/are-you-based-in-uk")
+        link.attr("target") mustBe "_self"
+      }
     }
 
-    "display check CDS application status link" in {
-      val link = view.getElementById("check_cds_application_status_link")
+    "TdrUnauthorisedMessage is Enabled" should {
+      "display page header" in {
+        val view: Html = getUnauthorisedPageView(true)
+
+        view.getElementsByTag("h1").first() must containMessage("unauthorised.tdr.heading")
+      }
+
+      "display contact email address link" in {
+        val view: Html = getUnauthorisedPageView(true)
+        val link = view.getElementById("contact_support_link")
 
-      link must containMessage("unauthorised.paragraph.2.link")
-      link must haveHref("https://www.tax.service.gov.uk/customs/register-for-cds/are-you-based-in-uk")
-      link.attr("target") mustBe "_self"
+        link must containMessage("unauthorised.tdr.body.link")
+        link must haveHref(s"mailto:${messages("unauthorised.tdr.body.link")}")
+        link.attr("target") mustBe "_blank"
+      }
     }
   }
+
+  private def getUnauthorisedPageView(tdrEnabled: Boolean = false) = {
+    when(tdrUnauthorisedMsgConfig.isTdrUnauthorisedMessageEnabled).thenReturn(tdrEnabled)
+    val unauthorisedPage = injector.instanceOf[unauthorised]
```

Comment:
This definition could be changed as a class field. Any reason to have it here?

Reply:
Not sure I understand, it is a method as it takes a dynamic parameter, how could it become an immutable class field?

Reply:
because the actual view's content is only built when the view's `apply` method is called and this takes place at the next line (96).